stackit/internal/services/iaas/networkarea/resource.go

```diff
@@ -0,0 +1,678 @@
+				},
+			},
+			"project_count": schema.Int64Attribute{
+				Description: "The amount of projects currently referencing this area.",
+				Computed:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(0),
+				},
+			},
+			"default_nameservers": schema.ListAttribute{
+				Description: "List of DNS Servers/Nameservers.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"network_ranges": schema.SetAttribute{
+				Description: "List of Network ranges.",
+				ElementType: types.StringType,
+				Required:    true,
+				Validators: []validator.Set{
+					setvalidator.SizeAtLeast(1),
+					setvalidator.SizeAtMost(64),
+				},
+			},
+			"transfer_network": schema.StringAttribute{
+				Description: "Classless Inter-Domain Routing (CIDR).",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"default_prefix_length": schema.Int64Attribute{
+				Description: "The default prefix length for networks in the network area.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(24),
+					int64validator.AtMost(29),
+				},
+				Default: int64default.StaticInt64(25),
+			},
+			"max_prefix_length": schema.Int64Attribute{
+				Description: "The maximal prefix length for networks in the network area.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(24),
+					int64validator.AtMost(29),
+				},
+				Default: int64default.StaticInt64(29),
+			},
+			"min_prefix_length": schema.Int64Attribute{
+				Description: "The minimal prefix length for networks in the network area.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(22),
+					int64validator.AtMost(29),
+				},
+				Default: int64default.StaticInt64(24),
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *networkAreaResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve values from plan
+	var model Model
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := model.OrganizationId.ValueString()
+	ctx = tflog.SetField(ctx, "organization_id", organizationId)
+
+	// Generate API request body from model
+	payload, err := toCreatePayload(ctx, &model)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area", fmt.Sprintf("Creating API payload: %v", err))
+		return
+	}
+
+	// Create new network area
+	var httpResp *http.Response
+	ctxWithHTTPResp := runtime.WithCaptureHTTPResponse(ctx, &httpResp)
+	area, err := r.client.CreateNetworkArea(ctxWithHTTPResp, organizationId).CreateNetworkAreaPayload(*payload).Execute()
```

Comment:
Do we need to capture the raw HTTP response? I think it is not used anywhere here

Reply:
@joaopalet -> It is only used in create network area function, nowhere else. 

Should we also remove raw resp from the `network` resource? https://github.com/stackitcloud/terraform-provider-stackit/blob/af7d7899455de0348679c91d05cc95b18f0e788b/stackit/internal/services/iaas/network/resource.go#L186C20-L186C33

Reply:
There we need it to get the `x-request-id` header, that is needed to get more details about the network